rs/ledger_suite/common/ledger_core/src/approvals.rs

```diff
@@ -474,3 +480,40 @@ where
 fn remote_future() -> TimeStamp {
     TimeStamp::from_nanos_since_unix_epoch(u64::MAX)
 }
+
+impl<Tokens: Clone + Into<Nat> + TryFrom<Nat, Error = String>> Storable for Allowance<Tokens> {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        let mut buffer = vec![];
+        let amount: Nat = self.amount.clone().into();
+        amount
+            .encode(&mut buffer)
+            .expect("Unable to serialize amount");
```

Comment:
I was thinking of making use of the fact that `U64` in `ic-icrc1-tokens-u64` and `U256` in `ic-icrc1-tokens-u256` implement the `Storable` trait, and have a bounded size, but the problem seems to be that `ic_ledger_core::tokens::Tokens` used by the ICP ledger does implement `Storable`. Would it be possible (and make sense) to implement the `Storable` trait for `ic_ledger_core::tokens::Tokens` and use that? Then it should be possible to calculate a `max_size` based on the token type, and not use `Bound::Unbounded`. We should then also be able to support e.g., `u512`, since the bounded `max_size` would be adapted.

Reply:
Based on [this internal Slack discussion](https://dfinity.slack.com/archives/C0547JX2U48/p1729159169432729?thread_ts=1728991396.891529&cid=C0547JX2U48), there shouldn't be too much difference between bounded and unbounded, so I would just go with unbounded.

Reply:
Thank you for investigating this!